megatron/core/models/mamba/mamba_model.py

```diff
@@ -168,6 +194,94 @@ def set_input_tensor(self, input_tensor: Tensor) -> None:
         assert len(input_tensor) == 1, 'input_tensor should only be length 1 for gpt/bert'
         self.decoder.set_input_tensor(input_tensor[0])
 
+    def _preprocess(
```

Comment:
Is this an MTP-specific pre-process? Should MTP be in its name?

I'm also wondering if it would make the code easier to read if this was defined after `forward`.